### PR TITLE
rename std::string requires to demands

### PR DESCRIPTION
**Pull request by rwstrom**
This allows compilation with c++20 where 'requires' is now a keyword.

### CODE REVIEW

**Comment by rwstrom** [comment]
@googlebot I signed it!

**Comment by rwstrom** [comment]
@googlebot I fixed it.

File: galogen.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ struct TypeInfo {
   std::string type_cdecl;
   
   // Name of another type that this type requires.
-  std::string requires;
```
Comment (nicebyte) [inline galogen.cpp:48]: could you use `prerequisite_type` instead? now that i look at it, `requires` wasn't that good of a choice in the first place.

Comment (rwstrom) [reply galogen.cpp:48]: sure thing.

**Comment by nicebyte** [comment]
thanks!

**Pull request merged**